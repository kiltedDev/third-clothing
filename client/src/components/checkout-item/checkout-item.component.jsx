import React from 'react';
import { connect } from 'react-redux';

import {
  addItem,
  clearItemFromCart,
  removeItem
} from '../../redux/cart/cart.actions';

import './checkout-item.styles.scss';
import {
  CheckoutItemContainer,
  ImageContainer,
  TextContainer,
  QuantityContainer,
  RemoveButtonContainer
} from './checkout-item.styles';

export const CheckoutItem = ({ cartItem, clearItem, addItem, removeItem }) => {
  const { imageUrl, name, price, quantity } = cartItem;
  return(
    <CheckoutItemContainer className='checkout-item'>
      <ImageContainer
        className='image-container'
        children={
          <img src={ imageUrl } alt={ name } />
        }
      />
      <TextContainer className='name' children={ name } />
      <QuantityContainer className='quantity'>
        <div onClick={() => removeItem( cartItem ) } className='arrow left'>&#10094;</div>
        <span className='value'>{ quantity }</span>
        <div onClick={() => addItem( cartItem ) } className='arrow right'>&#10095;</div>
      </QuantityContainer>
      <TextContainer className='price' children={ `$${ price }` } />
      <RemoveButtonContainer
        className='remove-button'
        onClick={() => clearItem(cartItem)} >
        &#10005;
      </RemoveButtonContainer>
    </CheckoutItemContainer>
  )
}

const mapDispatchToProps = dispatch => ({
  clearItem: item => dispatch(clearItemFromCart(item)),
  addItem: item => dispatch(addItem(item)),
  removeItem: item => dispatch(removeItem(item))
})

export default connect(null, mapDispatchToProps)(CheckoutItem);
