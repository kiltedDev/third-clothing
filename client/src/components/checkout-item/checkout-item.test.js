import { shallow } from 'enzyme';
import React from 'react';
import { CheckoutItem } from './checkout-item.component';

describe('CheckoutItem', () => {
  let wrapper;
  let mockRemoveItem;
  let mockAddItem;
  let mockClearItem;

  beforeAll(() => {
    mockRemoveItem = jest.fn();
    mockAddItem = jest.fn();
    mockClearItem = jest.fn();

    const mockProps = {
      cartItem: {
        imageUrl: 'www.coolimage.com',
        name: 'Really Cool Coat',
        price: 25,
        quantity: 3
      },
      removeItem: mockRemoveItem,
      addItem: mockAddItem,
      clearItem: mockClearItem
    }
    wrapper = shallow(<CheckoutItem { ...mockProps } />)
  });

  it('should render CheckoutItem component',() => {
    expect( wrapper.debug() ).toMatchSnapshot();
  })

  it('should call removeItem when < is clicked',() => {
    wrapper.find('.quantity .left').simulate('click');
    expect(mockRemoveItem).toHaveBeenCalled();
  })

  it('should call addItem when > is clicked',() => {
    wrapper.find('.quantity .right').simulate('click');
    expect(mockAddItem).toHaveBeenCalled();
  })

  it('should call clearItem when X is clicked',() => {
    wrapper.find('RemoveButtonContainer').simulate('click');
    expect(mockClearItem).toHaveBeenCalled();
  })
})
