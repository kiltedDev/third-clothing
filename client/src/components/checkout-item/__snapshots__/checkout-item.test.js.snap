// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem should render CheckoutItem component 1`] = `
"<styled.div className=\\"checkout-item\\">
  <styled.div className=\\"image-container\\">
    <img src=\\"www.coolimage.com\\" alt=\\"Really Cool Coat\\" />
  </styled.div>
  <styled.span className=\\"name\\">
    Really Cool Coat
  </styled.span>
  <QuantityContainer className=\\"quantity\\">
    <div onClick={[Function: onClick]} className=\\"arrow left\\">
      ❮
    </div>
    <span className=\\"value\\">
      3
    </span>
    <div onClick={[Function: onClick]} className=\\"arrow right\\">
      ❯
    </div>
  </QuantityContainer>
  <styled.span className=\\"price\\">
    $25
  </styled.span>
  <RemoveButtonContainer className=\\"remove-button\\" onClick={[Function: onClick]}>
    ✕
  </RemoveButtonContainer>
</styled.div>"
`;
